### PR TITLE
[DOPS-2395]: Create jenkins library for rust projects

### PR DESCRIPTION
**Pull request by f33r0**
# Task

[DOPS-2395]: Create jenkins library for rust projects

## Changes

1. Updated pipeline using jenkins library
2. Moved build steps to bash file

## Author

Signed-off-by: name <zaporozhets@soramitsu.co.jp>

[DOPS-2395]: https://soramitsu.atlassian.net/browse/DOPS-2395?

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,95 +1,17 @@
...
-        cleanup { cleanWs() }
-    }
-}
\ No newline at end of file
+def pipeline = new org.rust.substratePipeline(steps: this,
+      parachain: true,
+      disableSecretScanner: false,
+      initSubmodules: false,
+      staticScanner: false,
+      codeCoverage: false,
```
Comment (wpi86) [inline Jenkinsfile]: Why?

File: Jenkinsfile [hunk 2]
```diff
@@ -1,95 +1,17 @@
...
-        }
-        cleanup { cleanWs() }
-    }
-}
\ No newline at end of file
+def pipeline = new org.rust.substratePipeline(steps: this,
+      parachain: true,
+      disableSecretScanner: false,
+      initSubmodules: false,
+      staticScanner: false,
```
Comment (wpi86) [inline Jenkinsfile]: Why?

File: Jenkinsfile [hunk 3]
```diff
@@ -1,95 +1,17 @@
-@Library('jenkins-library')
+@Library('jenkins-library@feature/dops-2395/rust_library') _
```
Comment (wpi86) [inline Jenkinsfile]:
```suggestion
@Library('jenkins-library') _
```

**Review by wpi86 — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (f33r0) [reply Jenkinsfile]: only for substrate

File: Jenkinsfile (hunk 1 above)
Comment (f33r0) [reply Jenkinsfile]: only for substrate

File: Jenkinsfile (hunk 3 above)
Comment (f33r0) [reply Jenkinsfile]: demo

**Review by wpi86 — APPROVED** [review]

**Review by C4tWithShell — APPROVED** [review]

**Review by alexkg — APPROVED** [review]

**Review by wpi86 — APPROVED** [review]

**Review by wpi86 — APPROVED** [review]

File: Jenkinsfile [hunk 4]
```diff
@@ -1,95 +1,16 @@
...
-            script{
-                gitNotify('main-CI', currentBuild.result, currentBuild.result)
-            }
-        }
-        cleanup { cleanWs() }
-    }
-}
\ No newline at end of file
+def pipeline = new org.rust.AppPipeline(steps: this,
+      disableSecretScanner: false,
```
Comment (Cre-eD) [inline Jenkinsfile]: it's also must be by default

File: Jenkinsfile [hunk 5]
```diff
@@ -1,95 +1,16 @@
...
+      initSubmodules: false,
+      secretScannerExclusion: '.*Cargo.toml\$|.*pr.sh\$',
+      dockerImageTags: ['develop': 'dev', 'master': 'latest'],
```
Comment (Cre-eD) [inline Jenkinsfile]: it's also must be by default

File: Jenkinsfile [hunk 6]
```diff
@@ -1,95 +1,16 @@
...
-            }
-        }
-        cleanup { cleanWs() }
-    }
-}
\ No newline at end of file
+def pipeline = new org.rust.AppPipeline(steps: this,
+      disableSecretScanner: false,
+      initSubmodules: false,
+      secretScannerExclusion: '.*Cargo.toml\$|.*pr.sh\$',
```
Comment (Cre-eD) [inline Jenkinsfile]: it's also must be by default

File: Jenkinsfile [hunk 7]
```diff
@@ -1,95 +1,16 @@
...
+      dockerImageTags: ['develop': 'dev', 'master': 'latest'],
+      envImageName: 'docker.soramitsu.co.jp/sora2/parachain-env:latest',
+      appImageName: 'docker.soramitsu.co.jp/sora2/parachain',
+      buildTestCmds: ['housekeeping/scripts/build.sh'],
+      codeCoverage: false,
```
Comment (Cre-eD) [inline Jenkinsfile:7]: we need to enable it but make sure it won't break CI

File: Jenkinsfile [hunk 8]
```diff
@@ -1,95 +1,16 @@
...
+      buildArtifacts: 'parachain_template_runtime.compact.compressed.wasm',
+      assignReviewers: true,
```
Comment (Cre-eD) [inline Jenkinsfile]: it's also must be by default

File: Jenkinsfile [hunk 9]
```diff
@@ -1,95 +1,16 @@
...
+      prStatusNotif: true
```
Comment (Cre-eD) [inline Jenkinsfile]: it's also must be by default

**Review by wpi86 — APPROVED** [review]

**Review by Cre-eD — APPROVED** [review]

**Review by baotn166 — APPROVED** [review]

**Pull request merged**